code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -1,18 +1,26 @@
 /obj/item/reagent_containers/food/drinks/cans
-	var/canopened = 0
+	var/canopened = FALSE
 	var/is_glass = 0
 	var/is_plastic = 0
+	var/shaken = 0
+	var/bursting = FALSE
+	var/burstchance = 0
 
 /obj/item/reagent_containers/food/drinks/cans/New()
 	..()
 	flags &= ~OPENCONTAINER
 
 /obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user)
-	if(canopened == 0)
-		playsound(loc,'sound/effects/canopen.ogg', rand(10,50), 1)
-		to_chat(user, "<span class='notice'>You open the drink with an audible pop!</span>")
-		canopened = 1
+	if(!canopened)
+		if(shaken)
+			fizzyopen()
+			return
+		playsound(loc, 'sound/effects/canopen.ogg', rand(10, 50), 1)
+		canopened = TRUE
 		flags |= OPENCONTAINER
+		desc += "\nIt has been opened."
```

Comment:
Instead of needing to add this in multiple locations, I would put this in `examine`:
```dm
/obj/item/reagent_containers/food/drinks/cans/examine(mob/user)
	. = ..()
	if(canopened)
		. += "It has been opened."
```